src/pugixml.cpp

```diff
@@ -6077,6 +6094,22 @@ namespace pugi
 		return true;
 	}
 
+	PUGI__FN bool xml_node::remove_children()
+	{
+		for (xml_node_struct* child = this->_root->first_child; child; child = child->next_sibling)
+		{
+			if (!_root || !child || child->parent != _root) return false;
```

Comment:
Why do we need this condition here?

Reply:
Initially, i copied everything the `remove_child` method. so this is not needed?